lib/web/apiserver_test.go

```diff
@@ -11260,3 +11260,161 @@ func Test_setEntitlementsWithLegacyLogic(t *testing.T) {
+			acceptHeader:              []string{"text/plain"},
+			expectedContentTypeHeader: "text/plain",
+		},
+		{
+			providedBody:              baselineTestBody,
+			expectedBody:              []byte(baselineTestBody),
+			acceptHeader:              []string{"text/plain;q=0.8"},
+			expectedContentTypeHeader: "text/plain",
+		},
+		{
+			providedBody:              baselineTestBody,
+			expectedBody:              []byte(baselineTestBody),
+			acceptHeader:              []string{"text/*"},
+			expectedContentTypeHeader: "text/plain",
+		},
+		{
+			providedBody:              baselineTestBody,
+			expectedBody:              []byte(baselineTestBody),
+			acceptHeader:              []string{"text/plain", "application/json"},
+			expectedContentTypeHeader: "text/plain",
+		},
+		{
+			providedBody:              baselineTestBody,
+			expectedBody:              jsonTestResponseBody,
+			acceptHeader:              []string{"application/json"},
+			expectedContentTypeHeader: "application/json",
+		},
+		{
+			providedBody:              baselineTestBody,
+			expectedBody:              jsonTestResponseBody,
+			acceptHeader:              []string{"application/json", "text/plain"},
+			expectedContentTypeHeader: "application/json",
+		},
+		{
+			providedBody:              baselineTestBody,
+			expectedBody:              jsonTestResponseBody,
+			acceptHeader:              []string{"unsupported", "application/json"},
+			expectedContentTypeHeader: "application/json",
+		},
+		{
+			providedBody:              baselineTestBody,
+			expectedBody:              []byte(baselineTestBody),
+			acceptHeader:              []string{"application/scim+json"},
+			expectedContentTypeHeader: "text/plain",
+		},
+		{
+			providedBody:              baselineTestBody,
+			expectedBody:              []byte(baselineTestBody),
+			acceptHeader:              []string{"application/*"},
+			expectedContentTypeHeader: "text/plain",
+		},
+		{
+			providedBody:              baselineTestBody,
+			expectedBody:              []byte(baselineTestBody),
+			acceptHeader:              []string{"*/*"},
+			expectedContentTypeHeader: "text/plain",
+		},
+		{
+			providedBody:              baselineTestBody,
+			expectedBody:              []byte(baselineTestBody),
+			acceptHeader:              []string{"*"},
+			expectedContentTypeHeader: "text/plain",
+		},
+		{
+			providedBody:          baselineTestBody,
+			acceptHeader:          []string{"text/"}, // Malformed
+			errFunc:               require.Error,
+			shouldNotSendResponse: true,
+		},
+		{
+			providedBody:              baselineTestBody,
+			expectedBody:              []byte(baselineTestBody),
+			acceptHeader:              []string{"unsupportedtype"},
+			expectedContentTypeHeader: "text/plain",
+		},
+	}
+
+	for _, testCase := range testCases {
+		// Setup
+		req, err := http.NewRequest("", "", nil)
+		req.Header["Accept"] = testCase.acceptHeader
+		require.NoError(t, err)
+
+		respWriter := &serveTextResponseWriter{}
```

Comment:
Can you use `httptest.NewRecorder` as the responsewriter instead of making your own type?

Reply:
Oh that's great. I was looking for a built-in implementation but couldn't find one.